core/cache/redisBackend.go

```diff
@@ -0,0 +1,378 @@
+		BackendHitCounter
+		Pool             *redis.Pool
+		logger           flamingo.Logger
+		writeLockEnabled bool
+	}
+
+	// RedisCacheEntryMeta representation
+	RedisCacheEntryMeta struct {
+		Lifetime, Gracetime time.Duration
+	}
+
+	// RedisCacheEntry representation
+	RedisCacheEntry struct {
+		Meta RedisCacheEntryMeta
+		Data interface{}
+	}
+
+	// RedisBackendOptions ...
+	RedisBackendOptions struct {
+		Network string
+		// Redis Host (default 127.0.0.1)
+		Host string
+		// Redis Port (default 6379)
+		Port string
+		// Redis Database Number to use (default 0)
+		Db int
+		// Passwort for the rdis connection. empty string for none (default empty)
+		Password string
+		// Maximum number of idle redis connections in the pool. (default 8)
+		MaxIdle int
+		// Timout to close idle connections (default 5m)
+		IdleTimeout time.Duration
+		// WriteLock feature switch
+		WriteLock bool
+	}
+)
+
+const (
+	lockPrefix  = "lock:"
+	tagPrefix   = "tag:"
+	valuePrefix = "value:"
+)
+
+var (
+	redisKeyRegex = regexp.MustCompile(`[^a-zA-Z0-9]`)
+)
+
+func redisConnector(network, address, password string) (redis.Conn, error) {
+	c, err := redis.Dial(network, address)
+	if err != nil {
+		return nil, err
+	}
+	if password != "" {
+		if _, err := c.Do("AUTH", password); err != nil {
+			c.Close()
+			return nil, err
+		}
+	}
+	return c, err
+}
+
+func finalizer(b *RedisBackend) {
+	b.Close()
+}
+
+// DefaultRedisBackendOptions gets the default options for redis backend
+func DefaultRedisBackendOptions() RedisBackendOptions {
+	return RedisBackendOptions{
+		Network:     "tcp",
+		Host:        "127.0.0.1",
+		Port:        "6379",
+		Db:          0,
+		Password:    "",
+		MaxIdle:     8,
+		IdleTimeout: time.Minute * 30,
+		WriteLock:   false,
+	}
+}
+
+// NewRedisBackend creates an redis cache backend
+func NewRedisBackend(options RedisBackendOptions, logger flamingo.Logger) Backend {
+	err := mergo.Merge(&options, DefaultRedisBackendOptions())
+	if err != nil {
+		logger.WithField("category", "redisBackend").Error(fmt.Sprintf("Error merging options: %v", err))
+	}
+
+	b := &RedisBackend{
+		Pool: &redis.Pool{
+			MaxIdle:     options.MaxIdle,
+			IdleTimeout: options.IdleTimeout,
+			TestOnBorrow: func(c redis.Conn, t time.Time) error {
+				_, err := c.Do("PING")
+				return err
+			},
+			Dial: func() (redis.Conn, error) {
+				return redisConnector(options.Network, fmt.Sprintf("%v:%v", options.Host, options.Port), options.Password)
+			},
+		},
+		logger: logger,
+	}
+	b.withBackendType("redis")
+
+	runtime.SetFinalizer(b, finalizer) // close all connections on destruction
+
+	return b
+}
+
+// Inject RedisBackend dependencies
+func (b *RedisBackend) Inject(logger flamingo.Logger) {
+	b.logger = logger
+}
+
+// Close ensures all redis connections are closed
+func (b *RedisBackend) Close() {
+	b.Pool.Close()
+}
+
+// CreatePrefixedKey creates an redis-compatible key
+func (b *RedisBackend) CreatePrefixedKey(key string, prefix string) string {
```

Comment:
private?

Also I think prefixing should include cachename - if multiple caches use same redis.

I think the whole "factory" logic of creating frontend with backend(s) and the corresponding "Hitcounter" initialisation makes sense to have a short brainstorming on - because there are some invariant that makes sense to enforce.